src/flag_gems/ops/cumsum.py

```diff
@@ -48,6 +49,65 @@ def cumsum_kernel(
     tl.store(out_ptrs, result, mask=mask)
 
 
+
+@libentry()
+@triton.autotune(
+    configs=[
+        triton.Config({"BLOCK_M": 8, "BLOCK_N": MAX_C_MLU_CUMSUM//4}, num_warps=8, num_stages=4),
+        triton.Config({"BLOCK_M": 8, "BLOCK_N": MAX_C_MLU_CUMSUM//4}, num_warps=8, num_stages=5),
+        triton.Config({"BLOCK_M": 16, "BLOCK_N": MAX_C_MLU_CUMSUM//4}, num_warps=8, num_stages=4),
+        triton.Config({"BLOCK_M": 16, "BLOCK_N": MAX_C_MLU_CUMSUM//4}, num_warps=8, num_stages=5),
+        triton.Config({"BLOCK_M": 32, "BLOCK_N": MAX_C_MLU_CUMSUM//4}, num_warps=8, num_stages=4),
+        triton.Config({"BLOCK_M": 32, "BLOCK_N": MAX_C_MLU_CUMSUM//4}, num_warps=8, num_stages=5),
+        triton.Config({"BLOCK_M": 8, "BLOCK_N": MAX_C_MLU_CUMSUM//2}, num_warps=8, num_stages=4),
+        triton.Config({"BLOCK_M": 8, "BLOCK_N": MAX_C_MLU_CUMSUM//2}, num_warps=8, num_stages=5),
+        triton.Config({"BLOCK_M": 16, "BLOCK_N": MAX_C_MLU_CUMSUM//2}, num_warps=8, num_stages=4),
+        triton.Config({"BLOCK_M": 16, "BLOCK_N": MAX_C_MLU_CUMSUM//2}, num_warps=8, num_stages=5),
+        triton.Config({"BLOCK_M": 32, "BLOCK_N": MAX_C_MLU_CUMSUM//2}, num_warps=8, num_stages=4),
+        triton.Config({"BLOCK_M": 32, "BLOCK_N": MAX_C_MLU_CUMSUM//2}, num_warps=8, num_stages=5),
+        triton.Config({"BLOCK_M": 8, "BLOCK_N": MAX_C_MLU_CUMSUM}, num_warps=8, num_stages=4),
+        triton.Config({"BLOCK_M": 8, "BLOCK_N": MAX_C_MLU_CUMSUM}, num_warps=8, num_stages=5),
+        triton.Config({"BLOCK_M": 16, "BLOCK_N": MAX_C_MLU_CUMSUM}, num_warps=8, num_stages=4),
+        triton.Config({"BLOCK_M": 16, "BLOCK_N": MAX_C_MLU_CUMSUM}, num_warps=8, num_stages=5),
+        triton.Config({"BLOCK_M": 32, "BLOCK_N": MAX_C_MLU_CUMSUM}, num_warps=8, num_stages=4),
+        triton.Config({"BLOCK_M": 32, "BLOCK_N": MAX_C_MLU_CUMSUM}, num_warps=8, num_stages=5),
+    ],
+    key=[
+        "M",
+        "N",
+    ],
+)
+@triton.jit
+def cumsum_kernel_split(
+    inp,
+    out,
+    M,
+    N,
+    K,
+    BLOCK_M: tl.constexpr,
+    BLOCK_N: tl.constexpr,
+):
+    pid_m = tl.program_id(0)
+    pid_k = tl.program_id(1)
+    m_offset = pid_m * BLOCK_M + tl.arange(0, BLOCK_M)
+
+    kep = tl.full([BLOCK_M], float(0), tl.float32)
+    for col_offset in range(0, N, BLOCK_N):
+        n_offset = col_offset + tl.arange(0, BLOCK_N)
+        offset = m_offset[:, None] * N * \
+            K + n_offset[None, :] * K + pid_k
+        mask = m_offset[:, None] < M and n_offset[None, :] < N
+        inp_ptrs = inp + offset
+        inp_vals = tl.load(inp_ptrs, mask=mask, other=float(0)).to(tl.float32)
+
+        raw_res = tl.cumsum(inp_vals, axis=1)
+        result = raw_res + kep[:, None]
+        kep = result[:, BLOCK_N-1]
```

Comment:
does triton support tensor slice?

Reply:
I have the same question here. Tensor slicing of this type is not supported in triton 2.2.